persistence/sql/persister_consent.go

```diff
@@ -438,3 +438,92 @@ func (p *Persister) VerifyAndInvalidateLogoutRequest(ctx context.Context, verifi
 		return nil
 	})
 }
+
+func (p *Persister) FlushInactiveLoginConsentRequests(ctx context.Context, notAfter time.Time) error {
+	/* #nosec G201 table is static */
+	var lr consent.LoginRequest
+	var lrh consent.HandledLoginRequest
+
+	var cr consent.ConsentRequest
+	var crh consent.HandledConsentRequest
+
+	// Delete all entries (and their FK)
+	// where hydra_oauth2_authentication_request were timed-out or rejected
+	// - hydra_oauth2_authentication_request_handled
+	// - hydra_oauth2_consent_request_handled
+	// AND
+	// - hydra_oauth2_authentication_request.requested_at < ttl.login_consent_request
+	// - hydra_oauth2_authentication_request.requested_at < notAfter
+
+	// Using NOT EXISTS instead of LEFT JOIN or NOT IN due to
+	// LEFT JOIN not supported by Postgres and NOT IN will have performance hits with large tables.
+	// https://stackoverflow.com/questions/19363481/select-rows-which-are-not-present-in-other-table/19364694#19364694
+	// Cannot use table aliasing in MYSQL, will work in Postgresql though...
+	err := p.Connection(ctx).RawQuery(fmt.Sprintf(`
+	DELETE
+	FROM %s
+	WHERE NOT EXISTS
+		(
+		SELECT NULL
```

Comment:
Is `NULL == NOT EXISTS` or `NULL == EXISTS`?

Reply:
The `Select NULL` part is irrelevant and just there as a place holder for MYSQL - https://dev.mysql.com/doc/refman/8.0/en/exists-and-not-exists-subqueries.html. 

In PostgreSQL does not need anything in the Select to work. See this SO thread https://stackoverflow.com/questions/19363481/select-rows-which-are-not-present-in-other-table/19364694#19364694.